src/node_options.cc

```diff
@@ -128,7 +128,12 @@ void PerIsolateOptions::HandleMaxOldSpaceSizePercentage(
   }
 
   // Get available memory in bytes
+#ifdef V8_COMPRESS_POINTERS
+  // When pointer compression is enabled, V8 uses a 4 GiB heap limit.
+  uint64_t total_memory = kMaxPointerCompressionHeap;  // 4 GiB
```

Comment:
What would happen here if the available physical memory was _less_ than 4gb? Not sure exactly how this is used, but this value _could_ be _larger_ than the available physical memory, could it not?

Reply:
Ah, you know, that's a great point. I had made the assumption that the constrained memory API would return that but it won't. We'll need to make this the lesser of the available memory or the max.